src/DIRAC/WorkloadManagementSystem/Service/JobStateUpdateHandler.py

```diff
@@ -372,26 +372,17 @@ def export_setJobParameters(cls, jobID, parameters):
     """ Set arbitrary parameters specified by a list of name/value pairs
         for job specified by its JobId
     """
-    failed = False
-
     if cls.elasticJobParametersDB:
-      for key, value in parameters:  # FIXME: should use a bulk method
-        res = cls.elasticJobParametersDB.setJobParameter(jobID, key, value)
-        if not res['OK']:
-          gLogger.error('Failed to add Job Parameters to cls.elasticJobParametersDB', res['Message'])
-          failed = True
-          message = res['Message']
+      res = cls.elasticJobParametersDB.setJobParameters(jobID, parameters)
+      if not res['OK']:
+        gLogger.error('Failed to add Job Parameters to ElasticJobParametersDB', res['Message'])
 
     else:
       result = cls.jobDB.setJobParameters(int(jobID), parameters)
       if not result['OK']:
         gLogger.error('Failed to add Job Parameters to MySQL', result['Message'])
-        failed = True
-        message = result['Message']
 
-    if failed:
-      return S_ERROR(message)
-    return S_OK()
+    return res
```

Comment:
in `not cls.elasticJobParametersDB` (aka. `else`) it is `result` and not res.
```python
result = cls.jobDB.setJobParameters(int(jobID), parameters)
```
Need to unify this

Reply:
oops! Thanks. Fixed